### ISSUE TITLE
How to extend array-type values using override insteadof concat?

### ISSUE THREAD

**Issue by dickeylth** [unclechu/node-deep-extend#18]
```
var extendedObj = deepExtend({
    a: [1,2,3]
}, {
    a: [2,3]
});
console.log(extendedObj);
// { a: [ 2, 3, 3 ] }, however in my situation I expect { a: [ 2, 3 ] }
```

It seems that `deep-extend` uses replacement by index in array and concat two arrays, however in my situation I want to override the first array value, how to do that?

**Comment by NinjaTux** [comment]
@dickeylth not sure if this should be a functionality for a function like "deep extend", otherwise you are making the function aware about the content of the array.

**Comment by unclechu** [comment]
@dickeylth now it follows jQuery.extend() style, and jQuery do extend arrays like that:
```javascript
$.extend([1,2,3], [2,3])
Array [ 2, 3, 3 ]
```

**Comment by unclechu** [comment]
May be it's time to create some custom module without arrays extending.

**Comment by unclechu** [comment]
@dickeylth oh, it's looks like it isn't whole picture, jQuery overwrite arrays if they inside objects:
```javascript
$.extend({a: [1, 2, 3]}, {a: [2, 3]}).a
Array [ 2, 3 ]
```

**Comment by unclechu** [comment]
Fixed by 9ea374c54bdff7e42be163ae09670ea8cf3bbc7e

**Comment by dickeylth** [comment]
Cool!~Thank you!
